lib/DocumentManager.php

```diff
@@ -175,4 +265,34 @@ private function getOptionsResolver($eventName)
 
         return $resolver;
     }
+
+    /**
+     * If the exception is an instanceof `DocumentManagerException` set the
+     * document manager name that threw.
+     *
+     * @param \Exception $e
+     * @param string $context
+     */
+    private function processException(\Exception $exception, $context)
```

Comment:
Why not accepting only `DocumentManagerException`s, only catch `DocumentManagerException`s above, and remove the if in this method?